java-symbol-solver-model/src/main/java/me/tomassetti/symbolsolver/model/typesystem/ReferenceTypeUsage.java

```diff
@@ -292,7 +292,8 @@ protected boolean compareConsideringTypeParameters(ReferenceTypeUsage other) {
         }
         if (this.getQualifiedName().equals(other.getQualifiedName())) {
             if (this.parameters().size() != other.parameters().size()) {
-                throw new IllegalStateException();
+                // one have the type not specified so they are compatible
+                return true;
```

Comment:
When indexing the JavaParser with Coati right now I have 92 cases where something goes wrong. These are not only method calls so maybe we can expand the tests on the JavaParser source code to make sure that these cases are solved correctly as well. I'll take a closer look at the kind of things I'm asking the JavaSymbolSolver.